src/DSP/MoogFilter.h

```diff
@@ -0,0 +1,53 @@
+/*
+  ZynAddSubFX - a software synthesizer
+
+  Moog Filter.h - Several analog filters (lowpass, highpass...)
+  Copyright (C) 2018-2018 Mark McCurry
+  Author: Mark McCurry
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+*/
+
+#pragma once
+#include "Filter.h"
+
+namespace zyn {
+
+class MoogFilter:public Filter
+{
+    public:
+        //! @param Fq resonance, range [0.1,1000], logscale
+        MoogFilter(unsigned char Ftype, float Ffreq, float Fq,
+                unsigned int srate, int bufsize);
+        ~MoogFilter() override;
+        void filterout(float *smp) override;
+        void setfreq(float /*frequency*/) override;
+        void setfreq_and_q(float frequency, float q_) override;
+        void setq(float /*q_*/) override;
+        void setgain(float dBgain) override;
+        void settype(unsigned char type); //
+
+    private:
+        unsigned sr;
+        float gain;
+        unsigned char type;
+
+        float step(float x);
+
+        float tanhXdivX(const float x);
+        float tanhX(const float x);
+        float tan_2(const float x);
```

Comment:
Somewhat pedantic, so feel free to ignore, but if possible marking functions const is handy in case we're trying to get info like filter response from a const object. Not really an issue though.

Reply:
it doesnt seem to care.
zynaddsubfx-code/src/DSP/MoogFilter.h:40:9: warning: type qualifiers ignored on function return type [-Wignored-qualifiers]
40 |         const float tanhXdivX(const float x);

looks like in C11 mode, type qualifiers are completely ignored on function return types

https://gcc.gnu.org/bugzilla/show_bug.cgi?id=39985#c5
https://gcc.gnu.org/git/?p=gcc.git&a=commit;h=aa4b467b680f230ab11922d1e29695e1eaba12af

Reply:
To be specific I mean "float tanhX(float x) const" not "const float tanhX(float x)" or "float tanhX(const float x)"

Reply:
ok